networkit/matching.pyx

```diff
@@ -0,0 +1,57 @@
+from _NetworKit import Matcher, PathGrowingMatcher
```

Comment:
Why are these declarations in the `.pyx` and not in the `.pxd`? Do we have a rationale for that? If not, we should probably write up a guideline. "Public" declarations go into the `.pxd` while implementation details go into `.pyx`?

Note that `structures` has all of its declarations in the `.pxd` file.

Reply:
`Matcher` and `PathGrowingMatcher` are Python classes that were being imported from `_NetworKit.pyx`. Importing them in `matching.pxd` would have no effect in `matching.pyx`. However, in 6118db8fa91a95dcad9656a50950e3b46eba28d8  `matching` is finalised and all imports are in the `pxd` file.

Reply:
So the rationale is: Python imports in .pyx, Cython imports in .pxd (and the latter become automatically visible in the .pyx)?